### PR TITLE
Update Generic Spec Tests to v0.9

### PR DESCRIPTION
**Pull request by rauljordan**
This PR builds on top of the logic from @0xKiwi to update ssz spec tests to v0.9.

**NOTE**: Certain "invalid" ssz check tests are skipped and will be fixed in another follow-up PR. See: #99

### CODE REVIEW

File: ssz.go [hunk 1]
```diff
@@ -128,6 +138,9 @@ func HashTreeRoot(val interface{}) ([32]byte, error) {
 	if val == nil {
 		return [32]byte{}, errors.New("untyped nil is not supported")
 	}
+	if b, ok := val.(bitfield.Bitvector4); ok {
```
Comment (shayzluf) [inline ssz.go]: why is bitvector a special case?

File: ssz_test.go [hunk 2]
```diff
@@ -767,6 +767,22 @@ func toBytes32(x []byte) [32]byte {
...
+}
+
+func TestBoolArrayExtraByte(t *testing.T) {
+	objBytes := hexDecodeOrDie(t, "01010101010101010101010101010101ff")
+	var result [16]bool
+	if err := Unmarshal(objBytes, &result); err == nil {
+		t.Fatal("expected message with extra byte to fail marshalling")
+	}
+}
+
```
Comment (shayzluf) [inline ssz_test.go]: maybe add a test for bool array that passes?

File: ssz.go (hunk 1 above)
Comment (rauljordan) [reply ssz.go]: You're right - it doesn't make much sense. I instead modified `HashTreeRootBitfield` to take in a generic interface and do a type assertion.

File: ssz_test.go (hunk 2 above)
Comment (rauljordan) [reply ssz_test.go]: Added

**Pull request merged**